compiler/rustc_trait_selection/src/solve/normalize.rs

```diff
@@ -253,20 +285,24 @@ impl<'tcx> TypeFolder<TyCtxt<'tcx>> for DeeplyNormalizeForDiagnosticsFolder<'_,
     }
 
     fn fold_ty(&mut self, ty: Ty<'tcx>) -> Ty<'tcx> {
-        deeply_normalize_with_skipped_universes(
-            self.at,
-            ty,
-            vec![None; ty.outer_exclusive_binder().as_usize()],
-        )
-        .unwrap_or_else(|_: Vec<ScrubbedTraitError<'tcx>>| ty.super_fold_with(self))
+        match deeply_normalize_with_skipped_universes_and_ambiguous_goals::<
+            _,
+            ScrubbedTraitError<'tcx>,
+        >(self.at, ty, vec![None; ty.outer_exclusive_binder().as_usize()])
+        {
+            Ok((value, _)) => value,
+            Err(_) => ty.super_fold_with(self),
+        }
     }
 
     fn fold_const(&mut self, ct: ty::Const<'tcx>) -> ty::Const<'tcx> {
-        deeply_normalize_with_skipped_universes(
-            self.at,
-            ct,
-            vec![None; ct.outer_exclusive_binder().as_usize()],
-        )
-        .unwrap_or_else(|_: Vec<ScrubbedTraitError<'tcx>>| ct.super_fold_with(self))
+        match deeply_normalize_with_skipped_universes_and_ambiguous_goals::<
+            _,
+            ScrubbedTraitError<'tcx>,
+        >(self.at, ct, vec![None; ct.outer_exclusive_binder().as_usize()])
+        {
+            Ok((value, _)) => value,
```

Comment:
We throw away ambiguous preds here b/c we may be using this folder on types that really are just not fully normalizable.

Reply:
The preds list should just contain stalled coroutine obligations, after all.